packages/fuel-gauge/src/await-execution.test.ts

```diff
@@ -0,0 +1,114 @@
+import { launchNode } from '@fuel-ts/wallet/test-utils';
+import {
+  Provider,
+  WalletUnlocked,
+  randomBytes,
+  Wallet,
+  BaseAssetId,
+  FUEL_NETWORK_URL,
+} from 'fuels';
+
+import { getSetupContract } from './utils';
+
+/**
+ * @group node
+ */
+describe('await-execution', () => {
+  test('awaiting execution of a transaction on the provider works', async () => {
+    const { cleanup, ip, port } = await launchNode({
+      args: ['--poa-interval-period', '400ms'],
+    });
+    const nodeProvider = await Provider.create(`http://${ip}:${port}/graphql`);
+
+    const genesisWallet = new WalletUnlocked(
+      process.env.GENESIS_SECRET || randomBytes(32),
+      nodeProvider
+    );
+
+    const destination = Wallet.generate({ provider: nodeProvider });
+
+    const transfer = await genesisWallet.createTransfer(destination.address, 100, BaseAssetId, {
+      gasPrice: nodeProvider.getGasConfig().minGasPrice,
+      gasLimit: 10_000,
+    });
+
+    transfer.updateWitnessByOwner(
+      genesisWallet.address,
+      await genesisWallet.signTransaction(transfer)
+    );
+
+    const response = await nodeProvider.sendTransaction(transfer, { awaitExecution: true });
+
+    expect(response.gqlTransaction?.status?.type).toBe('SuccessStatus');
+
+    cleanup();
+  });
+  test('calling contracts with awaitExecution works', async () => {
+    const contract = await getSetupContract('coverage-contract')();
+    const sendTransactionSpy = vi.spyOn(contract.provider, 'sendTransaction');
+    const gasPrice = contract.provider.getGasConfig().minGasPrice;
+
+    const { value } = await contract.functions
+      .echo_u8(3)
+      .txParams({ gasPrice, gasLimit: 10_000 })
+      .call({ awaitExecution: true });
```

Comment:
@nedsalk What is the practical difference between using the flag `awaitExecution` in the context of a contract call?

Even when we are not passing this flag as `true` and therefore not using `submitAndAwait`, the code will wait for the TX to be processed the same way.

Reply:
The practical difference is the number of calls towards the node:
- If you pass in `awaitExecution: false` or leave it `undefined`, the transaction will be submitted in one call and then awaited in another call via the [statusChange subscription in `TransactionResponse`](https://github.com/FuelLabs/fuels-ts/blob/78e5e405ef442b47032943196f67c3c0db125c92/packages/providers/src/transaction-response/transaction-response.ts#L212).
- If you pass in `awaitExecution: true`, the transaction will be submitted _and_ awaited in the **same** call.
- 
You touched upon a point I wanted to raise in our sync: given that the default behavior of contract function/script calls currently is to await the execution of the transaction, we can using this new `submitAndAwait` endpoint by default for them to reduce network calls until we flesh out what the actual default should be and how to signal that to the end user.

Reply:
@nedsalk I agree with you, I think we should use `submitAndAwait` by default when executing `call` on the `BaseInvocationScope`.

Reply:
@Torres-ssf  Implemented in https://github.com/FuelLabs/fuels-ts/commit/8e68217e693dfe888e9dda5bba3a732776e5e1e6.

Reply:
Given we are now using the subscription by default now, why do we still need to pass the `awaitExecution` config?

Reply:
@danielbate I covered it as it was a consumer of `provider.sendTransaction` where `awaitExecution` in the end resolves. Your question is valid and I now think that this PR shouldn't cover the `BaseInvocationScope.call` method. There were some discussions around what should be returned to the user when using `submitAndAwait` vs using the regular `submit` endpoint because one awaits until there are actual results for of the execution vs the other that returns a `TransactionResponse` of a transaction that could be in `SubmittedStatus` as well. In the end, I now believe that this `BaseInvocationScope.call` use case should be holistically covered by another PR.

Reply:
Gotcha, agreed.

Reply:
After contemplating a bit more on this, I decided to not introduce this feature on the account methods like `transfer` and `withdrawToBaseLayer` as well because it kind of suffers from the same DX dilemmas as `BaseInvocationScope.call`. I'd boil this PR down to a quick win for contract calls until I further investigate this DX rabbit hole and formulate a proper proposal.